test/tap/adduser-always-auth.js

```diff
@@ -46,19 +63,37 @@ test('npm login', function (t) {
       })
     })
 
-    var o = ''
-    var e = ''
-    var remaining = Object.keys(responses).length
-    runner.stdout.on('data', function (chunk) {
-      remaining--
-      o += chunk
-
-      var label = chunk.toString('utf8').split(':')[0]
-      runner.stdin.write(responses[label])
+    var message = 'Logged in as u on ' + common.registry + '/.'
+    runner.stdout.on('data', verifyStdout(runner, message, t))
+  })
+})
 
-      if (remaining === 0) runner.stdin.end()
+test('npm login --scope', function (t) {
```

Comment:
> what does the multiple argument `npm.config.get` do here (`npm.config.get('registry', 'cli')`)?

npm uses a package called [`config-chain`](http://npm.im/config-chain) to set up the precedence hierarchy between the various configuration files, environment variables, and configuration options set as command-line arguments. It's not really widely-known, but you can query one of those sources in particular by passing its name to `config.get()` as the second parameter, as you see here. The reason for this is that for the specific case of `npm login`, we don't want a `scope` set in a config file to trigger the scope-specific behavior for login.